src/Compilers/VisualBasic/Test/Semantic/Diagnostics/OperationAnalyzerTests.vb

```diff
@@ -1591,6 +1591,32 @@ End Class
                 Diagnostic(InvalidOperatorExpressionTestAnalyzer.InvalidUnaryDescriptor.Id, "-s").WithLocation(7, 16))
         End Sub
 
+        <WorkItem(9014, "https://github.com/dotnet/roslyn/issues/9014")>
+        <Fact>
+        Public Sub InvalidConstructorVisualBasic()
+            Dim source = <compilation>
+                             <file name="c.vb">
+                                 <![CDATA[
+Class C
+    Protected Structure S
+    End Structure
+End Class
+Class D
+    Shared Sub M(o)
+        M(New C.S())
+    End Sub
+End Class
+]]>
+                             </file>
+                         </compilation>
+
+            Dim comp = CompilationUtils.CreateCompilationWithMscorlibAndVBRuntime(source)
+            comp.VerifyDiagnostics(Diagnostic(ERRID.ERR_InaccessibleSymbol2, "C.S").WithArguments("C.S", "Protected").WithLocation(7, 15))
```

Comment:
I cannot point you to any specific helper (haven't been involved into analyzers testing), but, if there is one that uses base-line in the format similar to AssertTheseDiagnostics, it would be preferred.